### DISCUSSION TITLE
MediaElement.MetadataTitle from Audio stream (f.e. shoutcast radio)

### DISCUSSION THREAD

**Discussion by zaczy** [CommunityToolkit/Maui#2231] [New Feature Discussions]
Is it possible to automatic MediaElement.MetadataTitle setting from stream when source is radio stream with metadata included?

I see, that I can set property manually (for examply by quering shoutcast radio API in other task), but this information is also included in stream info, so it would be nice to fetch it automatically (and set property on demand)

**Comment by dodgex** [comment]
I am interested in this one too. A few weeks back I did some short testing with MAUI if i could replace React Native for an app I work on. But as old MAUI did not work with latest xcode back then I just came to retry by testing with dotnet/maui 10 and this MediaElement component.

But not having shoutcast radio metadata is a deal breaker, sadly. I am working on an app for an internet radio and rely on that information for track display during playback :D

**Reply by dodgex** [reply]
I starteda PoC for this, while I had success on Android and iOS (+Mac), it seems that unfortunately(?) it seems that the windows player does not provide/expose a way to access the metadata from the stream... 

Not sure if the MediaElement (or any other component in this toolkit) "allows" for `only supported on platform X, Y, Z` features.

**Reply by dodgex** [reply]
Also I have not tried or even looked at tizen support...

The changes I made can be found here https://github.com/dodgex/CommunityToolkit-Maui/tree/feat-icy-meta (please ignore the fact, that it is badly formatted and such :D)
